torch/csrc/autograd/python_mode.cpp

```diff
@@ -0,0 +1,44 @@
+#include <torch/csrc/autograd/python_mode.h>
+#include <torch/csrc/python_headers.h>
+#include <torch/csrc/autograd/python_variable.h>
+#include <ATen/PythonModeTLS.h>
+
+namespace torch { namespace autograd {
+
+PythonTorchDispatchTypeObject::PythonTorchDispatchTypeObject(
```

Comment:
Looking over the method invocations here, I don't think it's actually necessary to

Reply:
I'm not sure if I can use PyObject in TensorImpl.h. I could get it to work if we're okay with having it be void* instead

Reply:
we have a forward decl of PyObject for this very purpose :P  c.f. PyInterpreter

```
struct PyInterpreter;
struct C10_API PyInterpreter {
  using name_sig = std::string(const PyInterpreter*);
  using decref_sig = void(const PyInterpreter*, PyObject*);
  using detach_sig =

```

Reply:
Oh nice, let me go play around with that...